lib/Widgets/movie_builder_json.dart

```diff
@@ -0,0 +1,58 @@
+import 'dart:convert';
+import 'package:flutter/services.dart';
+
+import '../models/genres.dart';
+import '../models/movie.dart';
+
+class MovieBuilderJson {
+  static const String genresJsonPath = 'assets/data/genres.json';
+  static const String moviesJsonPath = 'assets/data/movies.json';
+  List<Movie> movies = [];
+  List<Genre> genres = [];
+
+  Future<List<Movie>> loadMoviesFromJson() async {
+    String genreData = await rootBundle.loadString(
+      genresJsonPath,
+    );
+    String movieData = await rootBundle.loadString(
+      moviesJsonPath,
+    );
+
+    List<dynamic> genresJson = jsonDecode(genreData)['genres'];
+    for (var genre in genresJson) {
+      genres.add(Genre.fromJson(
+        genre,
+      ));
+    }
+    List<dynamic> movieMap = jsonDecode(
+      movieData,
+    );
+    for (var movie in movieMap) {
+      List<dynamic> movieIds = movie['genre_ids'];
+      List<String> movieGenres = movieGenreCreator(
+        movieIds,
+      );
+      movies.add(
+        Movie.fromJson(
+          movie,
+          movieGenres,
+        ),
+      );
+    }
+    return movies;
+  }
+
+  List<String> movieGenreCreator(List<dynamic> ids) {
+    List<String> namesOfGenres = [];
+    for (int id in ids) {
+      for (Genre genre in genres) {
+        if (genre.id == id) {
+          namesOfGenres.add(
+            genre.name,
+          );
+        }
+      }
+    }
+    return namesOfGenres;
```

Comment:
This could be improved by using Dart's `Collection` methods:
https://medium.com/@darsshanNair/map-filter-where-and-reduce-in-dart-key-concepts-and-practical-usages-e5441dbc5b7

Reply:
Done!